load.go

```diff
@@ -237,6 +239,61 @@ func LoadSpec(dt []byte) (*Spec, error) {
 	return &spec, nil
 }
 
+// rawYAML is similar to json.RawMesage
+// We use this to store the raw yaml data for extension fields
+type rawYAML []byte
+
+func (y rawYAML) MarshalYAML() ([]byte, error) {
+	return y, nil
+}
+
+func (y *rawYAML) UnmarshalYAML(dt []byte) error {
+	*y = dt
+	return nil
+}
+
+func (f *extensionFields) UnmarshalYAML(dt []byte) error {
+	// We need to store the raw yaml data for each extension key.
+	// Parse the yaml, grab all the extension keys and store the raw values in f.
+
+	parsed, err := parser.ParseBytes(dt, parser.ParseComments)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing yaml: \n%s", string(dt))
+	}
+
+	if len(parsed.Docs) != 1 {
+		return errors.New("expected exactly one yaml document")
+	}
+
+	doc := parsed.Docs[0]
+	if doc.Body == nil {
+		return nil
+	}
+
+	body, ok := doc.Body.(*ast.MappingNode)
+	if !ok {
+		return errors.Errorf("expected a mapping node, got %T", body)
+	}
+
+	var ext extensionFields
+	for _, v := range body.Values {
+		key := v.Key.String()
+		if !strings.HasPrefix(key, "x-") && !strings.HasPrefix(key, "X-") {
```

Comment:
for my context, what's special about `x-` and `X-`?

Reply:
Dalec does strict unmarshalling, so if there are unknown fields it errors out.
We have specifically carved out top-level fields that start with `x-` or `X-`, which is a common way to deal with adding extra stuff to yaml files.

So this allows people to add `x-whatever` to their spec and not have unmarshalling error out.